src/main/java/com/clickhouse/kafka/connect/sink/state/provider/KeeperStateProvider.java

```diff
@@ -106,7 +112,15 @@ public StateRecord getStateRecord(String topic, int partition) {
             long maxOffset = r.getValue(2).asLong();
             State state = State.valueOf(r.getValue(3).asString());
             LOGGER.debug(String.format("read state record: topic %s partition %s with %s state max %d min %d", topic, partition, state, maxOffset, minOffset));
-            return new StateRecord(topic, partition, maxOffset, minOffset, state);
+
+            StateRecord stateRecord = new StateRecord(topic, partition, maxOffset, minOffset, state);
+            StateRecord storedRecord = stateMap.get(csc.getZkDatabase() + "-" + key);
+            if (storedRecord != null && !stateRecord.equals(storedRecord)) {
+                LOGGER.warn("State record is changed: {} -> {}", storedRecord, stateRecord);
```

Comment:
By design, it should change every time. if it changes correctly, I do not think we log it

Reply:
Let's also overload toString